src/libsyntax/diagnostic.rs

```diff
@@ -453,7 +453,7 @@ fn emit(dst: &mut EmitterWriter, cm: &codemap::CodeMap, rsp: RenderSpan,
             try!(highlight_lines(dst, cm, sp, lvl, cm.span_to_lines(sp)));
         }
     }
-    if sp != COMMAND_LINE_SP {
+    if sp.expn_id != COMMAND_LINE_EXPN && rsp.is_full_span() {
```

Comment:
There's some code a few lines up doing the same sort of `sp.expn_id` to `COMMAND_LINE_EXPN` comparison:
```
    // We cannot check equality directly with COMMAND_LINE_SP
    // since PartialEq is manually implemented to ignore the ExpnId
    let ss = if sp.expn_id == COMMAND_LINE_EXPN {
    ...
```
I'll factor out the `sp.expn_id == COMMAND_LINE_EXPN` to a local variable and put the comment there.